frontend/app/controllers/Joiner.scala

```diff
@@ -170,11 +183,24 @@ trait Joiner extends Controller with ActivityTracking {
       Future.sequence(optFuture.toSeq).map(_.headOption)
     }
 
+    val futureContentOpt = request.cookies.get("GU_MEM_REFERER").map(_.value).map { referer =>
+      val refererPath = referer.replace("http://www.theguardian.com", "")
+      contentApiService.contentItemQuery(refererPath).map(_.content.map(MembersOnlyContent))
+    }.getOrElse(Future.successful(None))
+
```

Comment:
Hah, I started out with something similar to this, until I trusted IntelliJ's `Refactor > Inline` option.